### PR TITLE
Exams by Stas Yukhymenko IM-23

### PR DESCRIPTION
**Pull request by stasyukhymenko**
I've completed 2 tasks:
1. sum.js
2. intersection.js

### CODE REVIEW

**Review by Nastia0703 — COMMENTED** [review]
Good job!

File: Tasks/sum.js [hunk 1]
```diff
@@ -1,20 +1,13 @@
...
-  let k = 5;
-  for (i of s) {
-    let t = typeof i;
-    if (t === 'number') {
-      if (sum.length > 0) {
-        const new_Sum = sum[sum.length - 1] + i;
-        sum.push(new_Sum);
-      }
-    }
+const sumOfArray = (args) => {
```
Comment (illiakroshka) [inline Tasks/sum.js:4]:
```suggestion
const sumOfArray = args => {
```
If there is one function parameter, you can pass it without brackets

File: Tasks/intersection.js [hunk 2]
```diff
@@ -1,25 +1,16 @@
...
-  }
-  {
+const intersection = (object1, object2) => {
+  const arrayOfKeys = Object.keys(object1);
+
+  for (const key of arrayOfKeys) {
+    const mutual =
+        object1[key] === object2[key]
+            ? (object2[key] = object1[key])
+            : delete object1[key];
```
Comment (tshemsedinov) [inline Tasks/intersection.js:11]: Use ternary only to branch calculations not to branch control flow, so do not put assign `=` operator and  `delete` operator inside ternary

File: Tasks/sum.js [hunk 3]
```diff
@@ -1,20 +1,13 @@
 // Sum all numbers from an array
+"use strict";
```
Comment (tshemsedinov) [inline Tasks/sum.js:2]: Check your code with eslint and prettier before commit

File: Tasks/sum.js [hunk 4]
```diff
@@ -1,20 +1,13 @@
...
-      if (sum.length > 0) {
-        const new_Sum = sum[sum.length - 1] + i;
-        sum.push(new_Sum);
-      }
-    }
+const sumOfArray = args => {
+  let result = 0;
+  const array = args.flat(Infinity);
+  for (const value of array) {
+    result = typeof value === "number" ? (result += value) : result;
```
Comment (tshemsedinov) [inline Tasks/sum.js:8]: Use if in this case

**Review by tshemsedinov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**